steve/util.py

```diff
@@ -155,17 +222,115 @@ def get(self, section, key):
     # api_key =
     """)
 
-    def create(self, dir_path):
-        abs_path = os.path.abspath(dir_path)
-        kw = dict(
-            projectpath=abs_path,
-            jsonpath=os.path.join(abs_path, 'json')
-        )
-        with open(os.path.join(dir_path, self.file_name), 'w') as fp:
-            fp.write(self._CONFIG.format(**kw))
 
+class YamlConfig(Config):
+    file_name = 'steve.yaml'
+
+    def __init__(self):
+        super(YamlConfig, self).__init__()
+
+    def load(self):
+        """Finds and opens the config file in the current directory
+
+        :raises ConfigNotFound: if the config file can't be found
+
+        :returns: config file
+
+        """
+        self._data = yaml_load(open(self.project_file))
+        project = self._data['project']
 
-SteveConfig = IniConfig   # to allow easy switching in the future
+        # TODO: This is a little dirty since we're inserting stuff into
+        # the config file if it's not there, but so it goes.
+        try:
+            project['projectpath']
+        except KeyError:
+            project['projectpath'] = self.project_path
+        try:
+            project['jsonpath']
+        except KeyError:
+            project['jsonpath'] = os.path.join(project['projectpath'], 'json')
+        try:
+            project['yamlpath']
+        except KeyError:
+            project['yamlpath'] = os.path.join(project['projectpath'], 'yaml')
+        # If STEVE_CRED_FILE is specified in the environment or there's a
+        # cred_file in the config file, then open the file and pull the
+        # API information from there:
+        #
+        # * api_url
+        # * username
+        # * api_key
+        #
+        # This allows people to have a whole bunch of steve project
+        # directories and store their credentials in a central location.
+        cred_file = None
+        try:
+            cred_file = os.environ['STEVE_CRED_FILE']
+        except KeyError:
+            try:
+                cred_file = project['cred_file']
+            except KeyError:
+                pass
+
+        if cred_file:
+            cred_file = os.path.abspath(cred_file)
+
+            if os.path.exists(cred_file):
+                cfp = ConfigParser.ConfigParser()
+                cfp.read(cred_file)
+                project['api_url'] = cfp.get('default', 'api_url')
+                project['username'] = cfp.get('default', 'username')
+                project['api_key'] = cfp.get('default', 'api_key')
+
+        return self
```

Comment:
This is repeat code from `IniConfig.load` plus this has items that's not in `IniConfig.load`. Seems like we should hoist default-adding to `Config`.